contribute/engineering/backend/instrumentation.md

```diff
@@ -0,0 +1,207 @@
+# Instrumenting Grafana
+
+Guidance, conventions and best practices for instrumenting Grafana using logs, metrics and traces.
+
+## Logs
+
+Logs are files that record events, warnings and errors as they occur within a software environment. Most logs include contextual information, such as the time an event occurred and which user or endpoint was associated with it.
+
+Use the _pkg/infra/log_ package to create a named structured logger. Example:
+
+```go
+import (
+  "fmt"
+
+  "github.com/grafana/grafana/pkg/infra/log"
+)
+
+logger := log.New("my-logger")
+logger.Debug("Debug msg")
+logger.Warning("Warning msg")
+logger.Info("Info msg")
+logger.Error("Error msg", "error", fmt.Errorf("BOOM))
+```
+
+### Naming conventions
+
+Consider naming the logger using lowercase characters, e.g. `log.New("my-logger")` using snake_case or kebab-case styling.
+
+Consider to prefix the logger name with an area name if using different loggers across a feature/common packages, e.g. `log.New("plugin.loader")` and `log.New("plugin.client")`.
+
+Start the log message with a capital letter, e.g. `logger.Info("Hello world")` instead of `logger.Info("hello world")`.
```

Comment:
```suggestion
Start the log message with a capital letter, e.g. `logger.Info("Hello world")` instead of `logger.Info("hello world")`. The log message should be an identifier for the log entry, avoid parameterization in favor of key-value pairs for additional data.
```